app/forms/floating-ip-create.tsx

```diff
@@ -0,0 +1,182 @@
+  const isPoolSelected = !!form.watch('pool')
+
+  const [openItems, setOpenItems] = useState<string[]>([])
+
+  return (
+    <SideModalForm
+      id="create-floating-ip-form"
+      title="Create Floating IP"
+      form={form}
+      onDismiss={() => navigate(pb.floatingIps(projectSelector))}
+      onSubmit={({ address, ...rest }) => {
+        createFloatingIp.mutate({
+          query: projectSelector,
+          // if address is '', evaluate as false and send as undefined
+          body: { address: address || undefined, ...rest },
+        })
+      }}
+      loading={createFloatingIp.isPending}
+      submitError={createFloatingIp.error}
+    >
+      <NameField name="name" control={form.control} />
+      <DescriptionField name="description" control={form.control} />
+
+      <Accordion.Root
+        type="multiple"
+        className="mt-12 max-w-lg"
+        value={openItems}
+        onValueChange={setOpenItems}
+      >
+        <AccordionItem
+          value="advanced"
+          label="Advanced"
+          isOpen={openItems.includes('advanced')}
+        >
+          <Message
+            variant="info"
+            content="If you don’t specify a pool, the default will be used."
+          />
+
+          <ListboxField
+            name="pool"
+            items={allPools.items.map((p) => toListboxItem(p))}
+            label="IP pool"
+            control={form.control}
+            placeholder="Select pool"
+          />
+          <TextField
+            name="address"
+            control={form.control}
+            disabled={!isPoolSelected}
+            transform={(v) => v.replace(/\s/g, '')}
+            validate={(ip) =>
+              ip && !validateIp(ip).valid ? 'Not a valid IP address' : true
+            }
+          />
+        </AccordionItem>
+      </Accordion.Root>
+    </SideModalForm>
+  )
+}
+
+type AccordionItemProps = {
+  value: string
+  isOpen: boolean
+  label: string
+  children: React.ReactNode
+}
+
+function AccordionItem({ value, label, children, isOpen }: AccordionItemProps) {
+  const contentRef = useRef<HTMLDivElement>(null)
+
+  useEffect(() => {
+    if (isOpen && contentRef.current) {
+      contentRef.current.scrollIntoView({ behavior: 'smooth' })
+    }
+  }, [isOpen])
+
+  return (
+    <Accordion.Item value={value}>
+      <Accordion.Header className="max-w-lg">
+        <Accordion.Trigger className="group flex w-full items-center justify-between border-t pt-2 text-sans-xl border-secondary [&>svg]:data-[state=open]:rotate-90">
+          <div className="text-secondary">{label}</div>
+          <DirectionRightIcon className="transition-all text-secondary" />
+        </Accordion.Trigger>
+      </Accordion.Header>
+      <Accordion.Content
+        ref={contentRef}
+        forceMount
+        className={cn('ox-accordion-content overflow-hidden py-4', { hidden: !isOpen })}
+      >
+        {children}
+      </Accordion.Content>
+    </Accordion.Item>
+  )
+}
```

Comment:
This could be simplified a bit if we decide we don't care about the `scrollIntoView` business. It's a lot more important on instance create because those accordions are at the bottom of a long form. In this case it's less like that the user will have to scroll. On the other hand, it is pretty nifty. My inclination now after seeing the code is to shared `AccordionItem` only. Trying to share a full Accordion component with the `openItems` stuff wired up would force us into a rather horrible props API.

https://github.com/oxidecomputer/console/assets/3612203/c51ddc0e-752b-47eb-a0e7-772d5cd5d04f

For completeness, here would be the diff for the simplified version. It works minus the scrolling — this is more the default way of using the Radix accordion component. But like I said I think it's better to keep the fanciness and share `AccordionItem`. You could move it into `app/components`.

```diff
diff --git a/app/forms/floating-ip-create.tsx b/app/forms/floating-ip-create.tsx
index d259820b..ff565b9e 100644
--- a/app/forms/floating-ip-create.tsx
+++ b/app/forms/floating-ip-create.tsx
@@ -87,8 +87,6 @@ export function CreateFloatingIpSideModalForm() {
   const form = useForm({ defaultValues })
   const isPoolSelected = !!form.watch('pool')
 
-  const [openItems, setOpenItems] = useState<string[]>([])
-
   return (
     <SideModalForm
       id="create-floating-ip-form"
@@ -108,17 +106,8 @@ export function CreateFloatingIpSideModalForm() {
       <NameField name="name" control={form.control} />
       <DescriptionField name="description" control={form.control} />
 
-      <Accordion.Root
-        type="multiple"
-        className="mt-12 max-w-lg"
-        value={openItems}
-        onValueChange={setOpenItems}
-      >
-        <AccordionItem
-          value="advanced"
-          label="Advanced"
-          isOpen={openItems.includes('advanced')}
-        >
+      <Accordion.Root type="single" className="mt-12 max-w-lg" collapsible>
+        <AccordionItem value="advanced" label="Advanced">
           <Message
             variant="info"
             content="If you don’t specify a pool, the default will be used."
@@ -148,20 +137,11 @@ export function CreateFloatingIpSideModalForm() {
 
 type AccordionItemProps = {
   value: string
-  isOpen: boolean
   label: string
   children: React.ReactNode
 }
 
-function AccordionItem({ value, label, children, isOpen }: AccordionItemProps) {
-  const contentRef = useRef<HTMLDivElement>(null)
-
-  useEffect(() => {
-    if (isOpen && contentRef.current) {
-      contentRef.current.scrollIntoView({ behavior: 'smooth' })
-    }
-  }, [isOpen])
-
+function AccordionItem({ value, label, children }: AccordionItemProps) {
   return (
     <Accordion.Item value={value}>
       <Accordion.Header className="max-w-lg">
@@ -170,11 +150,7 @@ function AccordionItem({ value, label, children, isOpen }: AccordionItemProps) {
           <DirectionRightIcon className="transition-all text-secondary" />
         </Accordion.Trigger>
       </Accordion.Header>
-      <Accordion.Content
-        ref={contentRef}
-        forceMount
-        className={cn('ox-accordion-content overflow-hidden py-4', { hidden: !isOpen })}
-      >
+      <Accordion.Content className="ox-accordion-content overflow-hidden py-4">
         {children}
       </Accordion.Content>
     </Accordion.Item>
```

Reply:
We talked about this on the phone because my above stream of consciousness was not clear — the short version would have been: on second thought, let's share `AccordionItem` with instance create.